### DISCUSSION TITLE
Does Neural Prophet takes only known future features?

### DISCUSSION THREAD

**Discussion by Msaleh87** [ourownstory/neural_prophet#765] [Q&A - forecasting best practices] [answered]
I have a training data set of 15 months. I have trained the model for 12 months. Now I want to include the temperature and other features in the model training. Do I need these features to have known values in the testing dataset? Or only up to the training data set (the 12 months).

Also, this might seem trivial, but how to add the feature? I remember reading through the documentation regarding aggressors.

Another question, I added the holidays in my model using this `m.add_country_holidays("US", mode="additive")`. Do you think I should make another feature for weekdays and weekends (like 1s for weekdays, and 0s for weekends), or does the model pick up on this automatically?

**Comment by Msaleh87** [comment, marked as answer]
It takes both future known features and past ones as well. Here are the links in the documentation:

Modeling Lagged Regressors: https://neuralprophet.com/html/lagged-regressors.html

Modeling Future Regressors: https://neuralprophet.com/html/future-regressors.html?highlight=future%20regressor

@ourownstory Thanks for sharing these links, a couple of questions:
- In the documentation, it is mentioned the following: This is the command to add the reggresor: `m = m.add_lagged_regressor(names='A')`
By setting the `only_last_value` argument of the a`dd_lagged_regressor function`, the user can specify either to use only the last known value of the regressor within the input window or else use the same number of lags as auto-regression. 
Does this mean that inside `m = m.add_lagged_regressor(names='A')` I could specify  `only_last_value`? and how to do /write it in the code? Also, I didn't get "the user can **specify either to use only the last known value of the regressor within the input window**, what does this mean? 
or else **use the same number of lags as auto-regression**" by lags is it the n_lags?

- should I make another feature for weekdays and weekends (like 1s for weekdays, and 0s for weekends), or does the model pick up on this automatically?

**Reply by ourownstory** [reply]
Re: `add_lagged_regressor` if you use the current version on Gihub, you can set the n_lags separately:
https://github.com/ourownstory/neural_prophet/blob/00028d161e03121c68678567a37807977a5ca11c/neuralprophet/forecaster.py#L419
if you leave it 'auto' it will be same as autoregression.

Re: weekday/end: If you have weekly seasonality enabled, that should capture it, but you can try to introduce this as an additional event variable.
